test/combobo/index.js

```diff
@@ -711,7 +711,7 @@ describe('Combobo', () => {
       let options = simpleBox.cachedOpts.length;
       assert.equal(options, simpleBox.cachedOpts.length);
       simpleBox.setOptions(`<div class='option'>new Option</div>`);
-      assert.equal(options + 2, simpleBox.cachedOpts.length);
+      assert.equal(options + 1, simpleBox.cachedOpts.length);
```

Comment:
This fixes the test that exposed the issue but I don't see a new test for the specific issue you are fixing (which I'm still a little unclear on)

Reply:
sometimes  the options is adding multiple times with setOptions() function, when we update cachedOpts automatically currentOpts are updating. then it is causing duplication. Hence we added a condition to check in the setOptions() function to check whether the currentOpts already contain the option or not, accordingly we are updating the currentOpts

Reply:
ok but I don't see a new test for what this fix is doing (according to the pr title "condition added to handle duplicate options..." so I'd expect to see a test _proving_ that combobo works as described)